easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1457,6 +1457,11 @@ def _parse_dependency(self, dep, hidden=False, build_only=False):
             if dependency['full_mod_name'].split('/')[-1].startswith('.'):
                 dependency['hidden'] = True
 
+            if 'name' not in dependency['external_module_metadata']:
+                dependency['name'] = dependency['short_mod_name']
```

Comment:
I don't think we should be shoving in values here like this...
And anyway, it's done in the wrong place (should be done where we pick up the metadata, not here).

I would rather fix the code that blindly assumes there's a name and version for every toolchain component, since that's the real issue here?

Reply:
Yes, the core of the problem is in det_toolchain_element_details in
if tc_dep['name'] == elem:

The EXTERNAL_MODULE ones don't have a name, nor do they have an ['external_module_metadata]['name'] unless there is an actual module.

So the only "name" that is available then is short_mod_name.
So I do think that the "best" place to set "name" is in _parse_dependency.
Other wise we'd need to hunt down every place that might have a problem with this.
Which we might have to anyway when ['external_module_metadata]['name'] does exist...

Thoughts on that?